File: frontend/src/game/entities/mutations/player-join-mutation.ts
import { BinaryEntity } from '../binary-entity';
import { SignedBinaryWriter } from '../data/signed-binary-writer';
import { MutationType } from './mutation-type';

export class PlayerJoinMutation extends BinaryEntity {
    constructor(readonly isReisen: boolean) {
        super();
    }

    appendToBinaryOutput(writer: SignedBinaryWriter): void {
        writer.writeByte(MutationType.JOIN);
        writer.writeBoolean(this.isReisen);
    }
}

File: backend/src/main/java/moe/mewore/rabbit/entities/mutations/PlayerJoinMutation.java
package moe.mewore.rabbit.entities.mutations;

import java.io.DataInput;
import java.io.IOException;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import moe.mewore.rabbit.data.SafeDataOutput;
import moe.mewore.rabbit.entities.BinaryEntity;

@Getter
@RequiredArgsConstructor
public class PlayerJoinMutation extends BinaryEntity {

    private final boolean isReisen;

    public static PlayerJoinMutation decodeFromBinary(final DataInput input) throws IOException {
        return new PlayerJoinMutation(input.readBoolean());
    }

    @Override
    public void appendToBinaryOutput(final SafeDataOutput output) {
        output.writeByte(MutationType.JOIN.getIndex());
        output.writeBoolean(isReisen);
    }
}